uktrade/data-hub-frontend
# Reactify link subsidiary page

## Description of change
The page for linking a company subsidiary has been reactified using the same pattern as #5777. I've also done some minor refactoring to reduce duplication between the three company `CollectionList` pages.

## Test instructions

Go to a Global HQ company and scroll down to the business details hierarchy section. It should be possible to manually add a subsidiary.

## Screenshots

<img width="974" alt="Screenshot 2023-07-06 at 13 36 50" src="https://github.com/uktrade/data-hub-frontend/assets/36161814/3a84378f-9269-4207-a9d3-660b7a4da3ec">


## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'
